app/assets/javascripts/discourse/app/components/user-menu/notifications-list.js

```diff
@@ -138,56 +144,67 @@ export default class UserMenuNotificationsList extends UserMenuItemsList {
     return content;
   }
 
-  dismissWarningModal() {
-    if (this.currentUser.unread_high_priority_notifications > 0) {
-      const modalController = showModal("dismiss-notification-confirmation");
-      modalController.set(
-        "confirmationMessage",
-        I18n.t("notifications.dismiss_confirmation.body.default", {
-          count: this.currentUser.unread_high_priority_notifications,
-        })
-      );
-      return modalController;
-    }
-  }
-
-  @action
-  dismissButtonClick() {
+  async modalCallback() {
```

Comment:
`modalCallback` isn't too descriptive, especially since we sometimes call it without showing the modal. Maybe something like `performDismiss()` instead?